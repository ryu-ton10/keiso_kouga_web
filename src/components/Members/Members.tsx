import React from 'react';
import '../../style.css';
import icon from '../../assets/icon.jpg';
import { Row, Col } from 'antd';

const Members: React.FC = () => {
    return (
        <>
            <div className="member-title">
                メンバー
            </div>
            <div className="member-cell">
                <Row gutter={12}>
                    <Col lg={4} xs={8} className="member-icon"><img src={String(icon)} alt="icon" /></Col>
                    <Col lg={8} xs={16} className="member-name"><p>とんと</p></Col>
                    <Col lg={4} xs={8} className="member-icon"><img src={String(icon)} alt="icon" /></Col>
                    <Col lg={8} xs={16} className="member-name"><p>つきー</p></Col>
                </Row>
            </div>
        </>
    );
}

export default Members;
