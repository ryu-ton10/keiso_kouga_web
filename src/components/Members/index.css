/* ========= メンバー紹介 =========== */
.member {
    position: relative;
}

.member-title {
    height: 180px;
    vertical-align: middle;
    background-repeat: no-repeat;
    background-position: center 50%;
    background-size: cover;
    background-image: url("../../assets/member.JPG");
}

.member-title-text {
    height: 100%;
    display: flex;
    flex-direction: column;
    align-items: left;
    justify-content: center;
}

.member-title-rgba {
    height: 100%;
    background: rgba(0, 0, 0, 0.5);
}

.member-title-jp {
    text-align: left;
    font-family: 'Hiragino Mincho Pro';
    font-size: 48pt;
    color: white;
    padding: 0 0 0 40px;
}

.member-title-en {
    text-align: left;
    font-family: 'Hiragino Mincho Pro';
    font-size: 14pt;
    color: white;
    padding: 0 0 0 40px;
}

.member-cell {
    max-height: 30%;
    padding: 40px 40px 50px 40px;
    text-align:center;
    background-color: rgba(245, 245, 245, 1);
}

.member-cell p {
    font-family: 'Hiragino Mincho Pro';
    font-size: 32pt;
    color: rgba(105, 105, 105, 1);
    padding: 20px 0px 0px 0px;
}

.member-cell img {
    max-width: 150px;
    max-height: 150px;
    border-radius: 50%;
}

.member-icon {
    padding: 30px 30px;
}

.member-name {
    padding: 30px 30px;
}

