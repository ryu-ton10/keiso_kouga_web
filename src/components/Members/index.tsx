import React from 'react';
import './index.css';
import { Row, Col, Image, Carousel } from 'antd';
import { TwitterCircleFilled } from '@ant-design/icons';

type Member = {
    id: string;
    name: string;
    image: string;
    works: string[];
    twitter_url: string;
    keywords: string[];
    camera: string;
}

type Props = {
    members: Member[];
}

const Members: React.FC<Props> = (props) => {

    const { members } = props;

    return (
        <div className="member">
            <div className="member-title">
                <div className="member-title-rgba">
                    <div className="member-title-text">
                        <p className="member-title-jp">紡ギ人</p>
                        <p className="member-title-en">Members</p>
                    </div>
                </div>
            </div>
            <div className="member-cell">
                {members.map((member) => {
                    const { name, image, works, twitter_url, camera } = member;
                    return (
                        <Row>
                            <Col xs={24} sm={12} md={10} lg={10} className="member-profile-left">
                                <img src={process.env.PUBLIC_URL + image} alt="icon" />
                                <Row>
                                    <p className="name">{name}</p>
                                    <a href={twitter_url} rel="noopener noreferrer" target="_blank"><TwitterCircleFilled /></a>
                                </Row>
                                <p>使用機材: {camera}</p>
                            </Col>
                            <Col xs={24} sm={12} md={14} lg={14} className="member-profile-right">
                                <p>{name}'s Gallery</p>
                                <Carousel autoplay>
                                    {works.map((work) => {
                                        return (
                                            <Image src={process.env.PUBLIC_URL + "Gallery/" + work} />
                                        );
                                    })}
                                </Carousel>
                            </Col>
                        </Row>
                    );
                })}
            </div>
        </div>
    );
}

export default Members;
