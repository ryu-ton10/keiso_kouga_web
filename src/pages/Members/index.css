/* ========= main =========== */
.member {
  position: relative;
  background-color: rgba(116, 139, 177, 1);
}

/* ========= 見出しセクション =========== */
.member-title {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.member-title-jp {
  text-align: center;
  font-family: 'Hiragino Mincho Pro';
  color: white;
  margin-bottom: 0;
}

.member-title-en {
  text-align: left;
  font-family: 'Hiragino Mincho Pro';
  color: white;
  margin-bottom: 0;
}

/* ========= 見出しのレスポンスデザイン =========== */
@media (min-width: 280px) {
  .member-title {
    height: 100px;
  }
  .member-title-jp {
    font-size: 24pt;
  }
  .member-title-en {
    font-size: 9pt;
  }
}

@media (min-width: 360px) {
  .member-title {
    height: 120px;
  }
  .member-title-jp {
    font-size: 28pt;
  }
  .member-title-en {
    font-size: 11pt;
  }
}

@media (min-width: 576px) {
  .member-title {
    height: 150px;
  }
  .member-title-jp {
    font-size: 36pt;
  }
  .member-title-en {
    font-size: 14pt;
  }
}

@media (min-width: 960px) {
  .member-cell {
    margin: 0 50px 0 50px;
  }
  .member-title {
    height: 180px;
  }
  .member-title-jp {
    font-size: 40pt;
  }
  .member-title-en {
    font-size: 16pt;
  }
}

@media (min-width: 1200px) {
  .member-cell {
    margin: 0 120px 0 120px;
  }
}

/* ========= メンバー一覧 =========== */
.member-cell {
  text-align: center;
  flex-direction: column;
  padding: 10px 10px 10px 10px;
}
