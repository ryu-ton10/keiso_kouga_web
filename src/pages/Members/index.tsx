import React from 'react';
import './index.css';
import MemberCard from 'components/MemberCard';
import { Member } from 'keisoKougaWebData';

type Props = {
  members: Member[];
}

function Members(props: Props) {

  const { members } = props;

  return (
    <div className="member">
      <div className="member-title">
        <p className="member-title-jp">紡ギ人</p>
        <p className="member-title-en">Members</p>
      </div>
      <div className="member-cell">
        {members.map((member, index) => {
          const { name, icon, works, twitter_url, camera } = member;
          return (
            <MemberCard
              key={index}
              name={name}
              icon={icon}
              works={works}
              twitter_url={twitter_url}
              camera={camera}
            />
          );
        })}
      </div>
    </div>
  );
}

export default Members;
