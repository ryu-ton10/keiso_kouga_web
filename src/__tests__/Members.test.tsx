import renderer from 'react-test-renderer';
import Members from 'pages/Members';

const dummyMembers = [
  {
    id: "0",
    name: "test",
    icon: {
        webp:"sample.webp",
        jpeg:"sample.jpeg",
    },
    works: [
      {
        id: "0",
        url: "sample.jpeg",
      },
    ],
    twitter_url: "http://sample.com/",
    keywords: ["hoge"],
    camera: "fuga"
  },
]
/* ---------------- snapshot test -------------------- */
it('コンポーネントが描画されること', () => {
  const component = renderer.create(
    <Members members={dummyMembers} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
