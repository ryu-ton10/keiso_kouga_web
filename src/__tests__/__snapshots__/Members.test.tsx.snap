// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`コンポーネントが描画されること 1`] = `
<div
  className="member"
>
  <div
    className="member-title"
  >
    <p
      className="member-title-jp"
    >
      紡ギ人
    </p>
    <p
      className="member-title-en"
    >
      Members
    </p>
  </div>
  <div
    className="member-cell"
  >
    <div
      className="member-card"
    >
      <div
        className="member-profile-left"
      >
        <picture>
          <source
            srcSet="sample.webp"
            type="image/webp"
          />
          <img
            alt="icon"
            src="sample.jpeg"
          />
        </picture>
        <p
          className="name"
        >
          test
        </p>
        <p>
          使用機材: 
          fuga
        </p>
        <a
          href="http://sample.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            aria-label="twitter"
            className="anticon anticon-twitter"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="twitter"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M928 254.3c-30.6 13.2-63.9 22.7-98.2 26.4a170.1 170.1 0 0075-94 336.64 336.64 0 01-108.2 41.2A170.1 170.1 0 00672 174c-94.5 0-170.5 76.6-170.5 170.6 0 13.2 1.6 26.4 4.2 39.1-141.5-7.4-267.7-75-351.6-178.5a169.32 169.32 0 00-23.2 86.1c0 59.2 30.1 111.4 76 142.1a172 172 0 01-77.1-21.7v2.1c0 82.9 58.6 151.6 136.7 167.4a180.6 180.6 0 01-44.9 5.8c-11.1 0-21.6-1.1-32.2-2.6C211 652 273.9 701.1 348.8 702.7c-58.6 45.9-132 72.9-211.7 72.9-14.3 0-27.5-.5-41.2-2.1C171.5 822 261.2 850 357.8 850 671.4 850 843 590.2 843 364.7c0-7.4 0-14.8-.5-22.2 33.2-24.3 62.3-54.4 85.5-88.2z"
              />
            </svg>
          </span>
        </a>
      </div>
      <div
        className="member-profile-right"
      >
        <div
          style={
            Object {
              "backgroundColor": "#000",
              "border": "0",
              "display": "block",
              "height": 337.92,
              "margin": "0",
              "overflow": "hidden",
              "padding": "0",
              "width": 512,
            }
          }
        >
          <div
            style={
              Object {
                "border": "0",
                "display": "block",
                "height": 337.92,
                "margin": "0",
                "overflow": "hidden",
                "padding": "0",
                "position": "absolute",
                "width": 512,
              }
            }
          >
            <div
              className="rsis-container"
              onClick={[Function]}
              style={
                Object {
                  "border": "0",
                  "display": "block",
                  "height": "100%",
                  "left": 0,
                  "margin": "0",
                  "padding": "0",
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            >
              <div
                className="rsis-image"
                onTransitionEnd={[Function]}
                style={
                  Object {
                    "backgroundImage": "url(sample.jpeg)",
                    "backgroundSize": "cover",
                    "height": "100%",
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "top": 0,
                    "transform": "translate3d(0%, 0px, 0px)",
                    "transition": "0.5s",
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
